Document keystone constructor config options

Vultraz opened the pull request:
I wanted to document the `Keystone` constructor's config options. Is there a good reference for the exact purpose of most of these?

MadeByMike commented on packages/keystone/README.md:
```diff
@@ -6,14 +6,41 @@ order: 1
+| ----------------------- | ---------- | ---------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `defaultAccess`         | `Object`   | `{}`       | FOO                                                                                                                                                         |
+| `adapters`              | `Array`    | `[]`       | FOO                                                                                                                                                         |
+| `adapter`               | `Object`   | `null`     | Required. The database storage adapter. See the [Adapter Framework](https://v5.keystonejs.com/keystone-alpha/keystone/lib/adapters/) page for more details. |
```
`null` is not really the default so I suggest Moving  required into the default col similar to other docs:

```suggestion
| `adapter`               | `Object`   |  Required     | The database storage adapter. See the [Adapter Framework](https://v5.keystonejs.com/keystone-alpha/keystone/lib/adapters/) page for more details. |
```

MadeByMike commented on packages/keystone/README.md:
```diff
@@ -6,14 +6,41 @@ order: 1
+
+### Config
+
+| Option                  | Type       | Default    | Description                                                                                                                                                 |
```
Can we remove `FOO` from all of these? I'll try and help with some descriptions today if I get time :)

MadeByMike commented:
Thanks so much for this!

Vultraz commented:
Dropped the `FOO`s, added a `require` line to the JS example, reordered the list a bit to group related options, and clarified the `secureCookies` default.

MadeByMike commented:
Looks great! I'm prepared to merge this and come back and add more detail later :)
